lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -5182,6 +5182,14 @@ bool runSpirvWriterPasses(Module *M, std::ostream *OS, std::string &ErrMsg,
   // been given; otherwise only perform regularization.
   bool WriteSpirv = OS != nullptr;
 
+  Triple TargetTriple(M->getTargetTriple());
+  if ((TargetTriple.getArch() == Triple::spirv32 ||
+       TargetTriple.getArch() == Triple::spirv64) &&
+      TargetTriple.getSubArch() != Triple::NoSubArch) {
+    // We cannot set the exact SPIR-V version the translator will emit, so warn.
+    errs() << "warning: SPIR-V version in target triple will be ignored\n";
```

Comment:
Ok, I wonder though if we could also recommend to use --max-version in this diagnostic? I assume this could be some sort of the workaround?

Reply:
Sure, I can add an extra line.